scrapy/spiders/crawl.py

```diff
@@ -109,9 +110,13 @@ def _errback(self, failure):
         rule = self._rules[failure.request.meta['rule']]
         return self._handle_failure(failure, rule.errback)
 
-    def _parse_response(self, response, callback, cb_kwargs, follow=True):
+    async def _parse_response(self, response, callback, cb_kwargs, follow=True):
         if callback:
             cb_res = callback(response, **cb_kwargs) or ()
+            if isinstance(cb_res, AsyncIterable):
+                cb_res = await collect_asyncgen(cb_res)
```

Comment:
Should we document that for CrawlSpider behavior of async callbacks is different from regular spiders? 

If I'm not mistaken, here the behavior is similar to Scrapy < 2.7, where the complete output of a callback is awaited before the processing started, while after https://github.com/scrapy/scrapy/pull/4978/ the output is processed as soon as it's sent.

Reply:
Makes sense, thanks for pointing it out. Please note that this is coherent with `CrawlSpider.process_results`, which is documented to receive a list (although said docs only mention it in the context of the [XMLFeedSpider](https://docs.scrapy.org/en/latest/topics/spiders.html#scrapy.spiders.XMLFeedSpider.process_results))